client/src/templates/Challenges/classic/Show.js

```diff
@@ -169,88 +180,222 @@ class ShowClassic extends Component {
-        <ReflexContainer orientation='vertical'>
-          <ReflexElement flex={1} {...this.resizeProps}>
-            <SidePanel
-              className='full-height'
-              description={description}
-              guideUrl={createGuideUrl(slug)}
-              instructions={instructions}
-              section={dasherize(blockName)}
-              title={blockNameTitle}
-              videoUrl={videoUrl}
-            />
-          </ReflexElement>
-          <ReflexSplitter propagate={true} {...this.resizeProps} />
-          <ReflexElement flex={1} {...this.resizeProps}>
-            {editors}
-          </ReflexElement>
-          {showPreview && (
-            <ReflexSplitter propagate={true} {...this.resizeProps} />
+      <Preview
+        className='full-height'
+        disableIframe={this.state.resizing}
+      />
+    );
+  }
+
+  renderDesktopLayout() {
+    const challengeFile = this.getChallengeFile();
+    return (
+      <ReflexContainer orientation='vertical'>
+        <ReflexElement flex={1} {...this.resizeProps}>
+          {this.renderInstructionsPanel({ showToolPanel: true })}
+        </ReflexElement>
+        <ReflexSplitter propagate={true} {...this.resizeProps} />
+        <ReflexElement flex={1} {...this.resizeProps}>
+          {challengeFile && (
+            <ReflexContainer key={challengeFile.key} orientation='horizontal'>
+              <ReflexElement
+                flex={1}
+                propagateDimensions={true}
+                renderOnResize={true}
+                renderOnResizeRate={20}
+                {...this.resizeProps}
+                >
+                {this.renderEditor()}
+              </ReflexElement>
+              <ReflexSplitter propagate={true} {...this.resizeProps} />
+              <ReflexElement
+                flex={0.25}
+                propagateDimensions={true}
+                renderOnResize={true}
+                renderOnResizeRate={20}
+                {...this.resizeProps}
+                >
+                {this.renderTestOutput()}
+              </ReflexElement>
+            </ReflexContainer>
           )}
-          {showPreview ? (
+        </ReflexElement>
+        {this.hasPreview() && (
+          <Fragment>
+            <ReflexSplitter propagate={true} {...this.resizeProps} />
             <ReflexElement flex={0.7} {...this.resizeProps}>
-              <Preview
-                className='full-height'
-                disableIframe={this.state.resizing}
-              />
+              {this.renderPreview()}
             </ReflexElement>
-          ) : null}
-        </ReflexContainer>
+          </Fragment>
+        )}
+      </ReflexContainer>
+    );
+  }
 
+  renderMobileLayout() {
+    const {
+      data: {
+        challengeNode: {
+          videoUrl
+        }
+      },
+      currentTab,
+      moveToTab
+    } = this.props;
+
+    const editorTabPaneProps = {
+      mountOnEnter: true,
```

Comment:
We've done it because MonacoEditor detects and sets its height on first render. As a result, when `editor` and `output` are rendered without this options they are hidden and MonacoEditor sets its height to 0.